server/user-groups.ts

```diff
@@ -0,0 +1,91 @@
+import {FS} from '../lib/fs';
+
+const Groups = new Map<string | ID, Auth>();
+
+export abstract class Auth extends Map<ID, GroupSymbol> {
+	save: () => void;
+	constructor() {
+		super();
+		this.save = () => null;
+	}
+	setGroup(user: User, group: GroupSymbol) {
+		if (!user) throw new Error(`No user passed to setGroup, use forceGroup instead.`);
+		super.set(user.id, group);
+		return this.save;
+	}
+	forceGroup(userid: ID, group: GroupSymbol) {
```

Comment:
Why did you rename this from `setOfflineGroup` which I personally think was clearer?

Reply:
I actually think overriding `set` is probably better, anyway. Let me fiddle with this.